File: Q3. Max Consecutive Ones/findMaxConsecutiveOnes.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function (nums) {
    let max = 0;
    let cnt = 0;

    for (let num of nums) {
        if (num === 0) {
            max = Math.max(cnt, max)
            cnt = num
        }
        cnt += num
    }

    return Math.max(max, cnt)
};

File: Q3. Max Consecutive Ones/findMaxConsecutiveOnes.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int ans = 0;
        int maxNum = 0;

        for(auto num : nums){
            if(num != 1){
                maxNum = max(maxNum, ans);
                ans = num;
            }
            ans += num;
        }
        maxNum = max(ans, maxNum);

        return maxNum;
    }
};